File: src/main/java/ru/serdtsev/homemoney/HmException.java
package ru.serdtsev.homemoney;

public class HmException extends RuntimeException {
  private final HmException.Code code;

  public final String getCode() {
    return this.code.name();
  }

  public HmException(HmException.Code code) {
    super();
    this.code = code;
  }

  public HmException(HmException.Code code, String message) {
    super(message);
    this.code = code;
  }

  public enum Code {
    WrongAuth,
    UnknownAccountTypeCode,
    UnknownMoneyTrnStatus,
    IdentifiersDoNotMatch,
    WrongAmount,
    UserIdCookieIsEmpty
  }

  @Override
  public String toString() {
    return "HmException{" +
        "code=" + code +
        "} " + super.toString();
  }
}

File: src/main/java/ru/serdtsev/homemoney/HmException.kt
package ru.serdtsev.homemoney

class HmException(private val code: HmException.Code) : RuntimeException() {
  enum class Code {
    WrongAuth,
    UnknownAccountTypeCode,
    UnknownMoneyTrnStatus,
    IdentifiersDoNotMatch,
    WrongAmount,
    UserIdCookieIsEmpty
  }

  fun getCode(): String {
    return code.name
  }
}
